pkg/cstorpool/v1alpha2/cstorpool.go

```diff
@@ -69,30 +90,81 @@ func (l *cspList) FilterUIDs(p ...predicate) []string {
 // listBuilder enables building a
 // list of csp instances
 type listBuilder struct {
-	list *cspList
+	list *CSPList
 }
 
 // ListBuilder returns a new instance of
 // listBuilder object
 func ListBuilder() *listBuilder {
-	return &listBuilder{list: &cspList{}}
+	return &listBuilder{list: &CSPList{}}
 }
 
 // WithUIDs builds a list of cstor pools
 // based on the provided pool UIDs
 func (b *listBuilder) WithUIDs(poolUIDs ...string) *listBuilder {
 	for _, uid := range poolUIDs {
-		obj := &apis.CStorPool{}
-		obj.SetUID(types.UID(uid))
-		item := &csp{object: obj}
-		b.list.items = append(b.list.items, item)
+		obj := &csp{&apis.CStorPool{}}
+		obj.object.SetUID(types.UID(uid))
+		b.list.items = append(b.list.items, obj)
+	}
+	return b
+}
+
+// WithUIDNodeMap builds a cspList based on the provided
+// map of uid and nodename
+func (b *listBuilder) WithUIDNodeMaps(UIDNodeMap map[string]string) *listBuilder {
+	for k, v := range UIDNodeMap {
+		obj := &csp{&apis.CStorPool{}}
+		obj.object.SetUID(types.UID(k))
+		obj.object.SetAnnotations(map[string]string{string(scheduleOnHostAnnotation): v})
```

Comment:
Can you confirm if this adds to existing annotation or replaces the existing annotation?
Logic should only add this annotation to existing ones.